src/modules/components/member/memberAvatar/memberAvatar.tsx

```diff
@@ -0,0 +1,63 @@
+import * as blockies from 'blockies-ts';
+import classNames from 'classnames';
+import type React from 'react';
+import { isAddress, type Hash } from 'viem';
+import { useEnsAddress, useEnsAvatar, useEnsName } from 'wagmi';
+import { Avatar, Spinner, type IAvatarProps } from '../../../../core';
+
+export interface IMemberAvatarProps extends Omit<IAvatarProps, 'fallback'> {
+    /**
+     * ENS name of the user to lookup avatar src.
+     */
+    ensName?: string;
+    /**
+     * 0x address of the user to look up ENS name and avatar src.
+     */
+    address?: string;
+    /**
+     * Direct URL src of the user avatar image to be rendered.
+     */
+    avatarSrc?: string;
+}
+
+export const MemberAvatar: React.FC<IMemberAvatarProps> = (props) => {
+    const { ensName, address, avatarSrc, size = 'sm', className, ...otherProps } = props;
+    const isValidAddress = address ? isAddress(address) : false;
+    const isValidENSName = ensName ? ensName.length >= 7 && ensName.endsWith('.eth') : false;
+
+    const { data: ensAddressData, isLoading: addressLoading } = useEnsAddress({
+        name: ensName,
+        query: { enabled: isValidENSName },
+    });
+    const resolvedAddress = isValidAddress ? address : ensAddressData;
+
+    const { data: ensNameData, isLoading: nameLoading } = useEnsName({
+        address: resolvedAddress as Hash,
+        query: { enabled: !!resolvedAddress },
+    });
+    const resolvedName = isValidENSName ? ensName : ensNameData;
+
+    const { data: ensAvatarData, isLoading: avatarLoading } = useEnsAvatar({
+        name: resolvedName as string,
+        query: { enabled: !!resolvedName },
+    });
+    const resolvedAvatarSrc = avatarSrc ? avatarSrc : ensAvatarData;
+
+    const blockiesSrc = resolvedAddress ? blockies.create({ seed: resolvedAddress }).toDataURL() : undefined;
+
+    const containerSizeClass = {
+        sm: 'w-6 h-6',
+        md: 'w-10 h-10',
+        lg: 'w-16 h-16',
+    }[size];
+
+    return (
+        <div className={classNames('flex shrink-0 items-center justify-center', containerSizeClass)}>
+            {avatarLoading || addressLoading || nameLoading ? (
+                <Spinner size={size} />
```

Comment:
As we discussed, the `AvatarMember` component itself won't render the `<Spinner />` component, but it will be rendered by the `InputAddress` we resolving ens/addresses. In order to render the "loading" state of the Avatar component (grey background with pulse animation) when resolving ens/addresses, we could update the `Avatar` component to accept a `isLoading` property that, when set, renders the component as loading